flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcWriter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.data;
+
+import org.apache.flink.types.Row;
+import org.apache.iceberg.data.orc.BaseOrcWriter;
+import org.apache.iceberg.orc.OrcValueWriter;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.storage.ql.exec.vector.ColumnVector;
+import org.apache.orc.storage.ql.exec.vector.StructColumnVector;
+import org.apache.orc.storage.ql.exec.vector.VectorizedRowBatch;
+
+public class FlinkOrcWriter extends BaseOrcWriter<Row> {
+
+  private FlinkOrcWriter(TypeDescription schema) {
+    super(schema);
+  }
+
+  public static OrcValueWriter<Row> buildWriter(TypeDescription fileSchema) {
+    return new FlinkOrcWriter(fileSchema);
+  }
+
+  @Override
+  protected Converter<Row> createStructConverter(TypeDescription schema) {
+    return new RowConverter(schema);
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void write(Row value, VectorizedRowBatch output) {
+    int row = output.size++;
```

Comment:
We avoid using return values from `++` expressions. That helps readability because statement order is clear.